pkg/trace/remoteconfighandler/remote_config_handler.go

```diff
@@ -57,6 +76,53 @@ func (h *RemoteConfigHandler) Start() {
 
 	h.remoteClient.Start()
 	h.remoteClient.Subscribe(state.ProductAPMSampling, h.onUpdate)
+	h.remoteClient.Subscribe(state.ProductAgentConfig, h.onAgentConfigUpdate)
+}
+
+func (h *RemoteConfigHandler) onAgentConfigUpdate(updates map[string]state.RawConfig) {
+	mergedConfig, err := state.MergeRCAgentConfig(h.remoteClient.UpdateApplyStatus, updates)
+	if err != nil {
+		return
+	}
+
+	if len(mergedConfig.LogLevel) > 0 {
+		// Get the current log level
+		var newFallback seelog.LogLevel
+		newFallback, err = pkglog.GetLogLevel()
+		if err == nil {
+			h.configState.FallbackLogLevel = newFallback.String()
+			var resp *http.Response
+			resp, err = http.Post(fmt.Sprintf(h.configSetEndpointFormatString, mergedConfig.LogLevel), "", nil)
+			if err == nil {
+				resp.Body.Close()
+				h.configState.LatestLogLevel = mergedConfig.LogLevel
+				pkglog.Infof("Changing log level of the trace-agent to %s through remote config", mergedConfig.LogLevel)
+			}
+		}
+	} else {
+		var currentLogLevel seelog.LogLevel
+		currentLogLevel, err = pkglog.GetLogLevel()
+		if err == nil && currentLogLevel.String() == h.configState.LatestLogLevel {
+			pkglog.Infof("Removing remote-config log level override of the trace-agent, falling back to %s", h.configState.FallbackLogLevel)
+			var resp *http.Response
+			resp, err = http.Post(fmt.Sprintf(h.configSetEndpointFormatString, h.configState.FallbackLogLevel), "", nil)
+			if err == nil {
+				resp.Body.Close()
+			}
```

Comment:
Let's make sure we log the errors

Reply:
We need to keep any error to update the config state a bit later, so the error is also logged later now